plugins/inputs/win_perf_counters/win_perf_counters.go

```diff
@@ -265,6 +265,11 @@ func (m *Win_PerfCounters) Gather(acc telegraf.Accumulator) error {
 					} else if metric.instance == s {
 						// Catch if we set it to total or some form of it
 						add = true
+					} else if strings.Contains(metric.instance, "#") && strings.HasPrefix(metric.instance, s) {
```

Comment:
In my opinion it is impossible, because telegraf creates for every configured instance and WPC combination a unique [query](https://github.com/bullshit/telegraf/blob/fix-1827-win_perf_counters/plugins/inputs/win_perf_counters/win_perf_counters.go#L165) and [handle](https://github.com/bullshit/telegraf/blob/fix-1827-win_perf_counters/plugins/inputs/win_perf_counters/win_perf_counters.go#L125) to [fetch](https://github.com/bullshit/telegraf/blob/fix-1827-win_perf_counters/plugins/inputs/win_perf_counters/win_perf_counters.go#L251]) the counter data.

for example:
```
[[inputs.win_perf_counters.object]]
  ObjectName = "Process"
  Counters = ["% Processor Time"]
  Instances = ["chrome","chrome#4"]
  Measurement = "win_chrome_cpu"
```

```
Valid: \Process(chrome)\% Processor Time
Valid: \Process(chrome#4)\% Processor Time
win_chrome_cpu,instance=chrome,objectname=Process,host=winston-pc Percent_Processor_Time=0.16016027331352234 1490095720000000000
win_chrome_cpu,instance=chrome#4,objectname=Process,host=winston-pc Percent_Processor_Time=0 1490095720000000000
```

for the instance "chrome" there are no problems, but because phd returns for the query `\Process(chrome#4)\% Processor Time` only `ch` as instance name the if statment at [win_perf_counters.go#L265](https://github.com/bullshit/telegraf/blob/fix-1827-win_perf_counters/plugins/inputs/win_perf_counters/win_perf_counters.go#L265) isn't working. Thats the reason why i check if the configured instance name contains an hash and, just to be sure, the returned name from phd starts like the configured name. If phd.dll will be fixed in the future my changes won't be executed because of the previous else if statment on line [win_perf_counters.go#L265](https://github.com/bullshit/telegraf/blob/fix-1827-win_perf_counters/plugins/inputs/win_perf_counters/win_perf_counters.go#L265). I don't see any reason why there should be a process prefix collision.